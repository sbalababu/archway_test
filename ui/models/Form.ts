export interface QuotaForm {
  quota: number;
}

export interface CoreMemoryForm {
  core: number;
  memory: number;
}

export interface MemberForm {
  username: string;
  roles: object;
}
