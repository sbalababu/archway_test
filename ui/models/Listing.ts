export interface Filters {
  filter: string;
  behaviors: string[];
  statuses: string[];
}
