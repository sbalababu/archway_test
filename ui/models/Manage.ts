import { ComplianceType } from '../constants';

export interface Question {
  question: string;
  requester: string;
  updated: Date;
  id?: number;
  complianceGroupId?: number;
}

export interface ComplianceContent {
  name: ComplianceType;
  id?: number;
  description: string;
  questions: Question[];
}

export interface Link {
  name: string;
  description: string;
  url: string;
  id?: number;
  customLinkGroupId?: number;
}

export interface LinksGroup {
  name: string;
  description: string;
  links: Link[];
  id?: number;
}
