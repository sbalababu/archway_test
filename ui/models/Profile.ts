export interface Profile {
  name: string;
  username: string;
  distinguished_name: string;
  permissions: {
    risk_management: boolean;
    platform_operations: boolean;
  };
}
