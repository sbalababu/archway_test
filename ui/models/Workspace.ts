export interface YarnApplication {
  id: string;
  name: string;
  start_time?: Date;
}

export interface ResourcePoolsInfo {
  loading: boolean;
  error?: string;
  data: PoolInfo[];
}

export interface PoolInfo {
  resource_pool: string;
  applications: YarnApplication[];
}

export interface NamespaceInfo {
  name: string;
  tables: HiveTable[];
}

export interface NamespaceInfoList {
  loading: boolean;
  error?: string;
  data: NamespaceInfo[];
}

export interface HiveTable {
  name: string;
}

export interface MemberRemoveStatus {
  loading?: boolean;
  success?: boolean;
  error?: boolean;
}

export interface Member {
  distinguished_name: string;
  email?: string;
  name: string;
  removeStatus?: MemberRemoveStatus;
  data: any;
  topics: any;
  applications: any;
}

export interface Application {
  id: number;
  name: string;
  consumer_group: string;
  group: SecurityGroup;
  type?: string;
  logo?: string;
  language?: string;
  repository?: string;
}

export interface TopicGrant {
  id: number;
  group: SecurityGroup;
  actions: string;
  topic_access?: Date;
}

export interface KafkaTopic {
  id: number;
  name: string;
  partitions: number;
  replication_factor: number;
  managing_role: TopicGrant;
  readonly_role: TopicGrant;
}

export interface ResourcePool {
  id: number;
  pool_name: string;
  max_cores: number;
  max_memory_in_gb: number;
}

export interface SecurityGroup {
  common_name: string;
  distinguished_name: string;
  sentry_role: string;
  group_created?: Date;
  role_created?: Date;
  role_associated?: Date;
  attributes?: string[][];
}

export interface Compliance {
  phi_data?: boolean;
  pci_data?: boolean;
  pii_data?: boolean;
}

export interface DatabaseGrant {
  group: SecurityGroup;
  location_access?: Date;
  database_access?: Date;
}

export interface HiveAllocation {
  id: number;
  name: string;
  location: string;
  size_in_gb: number;
  consumed_in_gb: number;
  managing_group: DatabaseGrant;
  readwrite_group?: DatabaseGrant;
  readonly_group?: DatabaseGrant;
}

export interface ApprovalStatus {
  loading?: boolean;
  success?: boolean;
  error?: string;
}

export interface ApprovalItem {
  approver?: string;
  approval_time?: Date;
  status?: ApprovalStatus;
}

export interface Approvals {
  infra?: ApprovalItem;
  risk?: ApprovalItem;
}

export interface Workspace {
  id: number;
  name: string;
  summary: string;
  description: string;
  behavior: string;
  requested_date: Date;
  requester: string;
  single_user: boolean;
  status?: string;
  compliance: Compliance;
  approvals?: Approvals;
  data: HiveAllocation[];
  processing: ResourcePool[];
  topics: KafkaTopic[];
  applications: Application[];
}

export interface UserSuggestion {
  display: string;
  distinguished_name: string;
}

export interface UserSuggestions {
  filter: string;
  users: UserSuggestion[];
  groups: UserSuggestion[];
}

export interface WorkspaceSearchResult {
  id: number;
  name: string;
  summary: string;
  status: string;
  behavior: string;
  pii_data?: boolean;
  pci_data?: boolean;
  phi_data?: boolean;
  date_requested: Date;
  date_fully_approved: Date;
  total_disk_allocated_in_gb: number;
  total_disk_consumed_in_gb?: number;
  total_max_cores: number;
  total_max_memory_in_gb: number;
}
