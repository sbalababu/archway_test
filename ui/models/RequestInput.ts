import { Compliance } from './Workspace';

export interface RequestInput {
  name: string;
  summary: string;
  description: string;
  compliance: Compliance;
}
