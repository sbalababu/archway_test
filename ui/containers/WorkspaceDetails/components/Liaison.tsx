import * as React from 'react';
import { Icon } from 'antd';
import { Colors } from '../../../components';
import { Member } from '../../../models/Workspace';

interface Props {
  data?: Member;
}

const Liaison = ({ data }: Props) =>
  data ? (
    <div style={{ display: 'flex', alignItems: 'center' }}>
      <div style={{ textTransform: 'uppercase', textAlign: 'right', marginRight: 8 }}>
        <div style={{ fontSize: 18, fontWeight: 300 }}>OWNER</div>
        <a href={`mailto:${data.email}`} style={{ color: Colors.Green.string(), textTransform: 'uppercase' }}>
          {data.name}
        </a>
      </div>
      <Icon style={{ fontSize: 40 }} type="crown" theme="twoTone" twoToneColor="#D7C9AA" />
    </div>
  ) : null;

export default Liaison;
