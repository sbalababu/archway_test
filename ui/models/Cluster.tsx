import { Dropdown, Icon, Menu } from 'antd';
import * as Color from 'color';
import * as React from 'react';
import { Colors } from '../components';

export interface WebLocation {
  host: string;
  port: number;
}

export interface Statusable {
  status: string;
}

export class Status<T extends Statusable> {
  public statusable: T;

  constructor(statusable: T) {
    this.statusable = statusable;
  }

  public statusColor = (): Color => {
    switch (this.statusable.status) {
      case 'GOOD_HEALTH':
        return Colors.Green;
      case 'CONCERNING_HEALTH':
        return Colors.Orange;
      case 'BAD_HEALTH':
        return Colors.Red;
      default:
        return Colors.Gray;
    }
  };

  public statusText = () => {
    switch (this.statusable.status) {
      case 'GOOD_HEALTH':
        return 'good';
      case 'CONCERNING_HEALTH':
        return 'concerning';
      case 'BAD_HEALTH':
        return 'bad';
      default:
        return 'unknown';
    }
  };

  public glowColorText = () =>
    `0 0 5px 2px ${this.statusColor()
      .hsl()
      .string()}`;
}

export interface HueService extends Statusable {
  load_balancer: WebLocation[];
}

/* tslint:disable:no-empty-interface */
export interface HiveService extends Statusable {
  thrift: WebLocation[];
}

export interface YarnService extends Statusable {
  resource_manager: WebLocation[];
  node_manager: WebLocation[];
}

export interface NavigatorService extends Statusable {
  navigator: WebLocation[];
}

export abstract class ServiceLinks<T extends Statusable> {
  public service: T;

  public links!: JSX.Element[];

  constructor(service: T) {
    this.service = service;
  }
}

export class HueServiceLinks extends ServiceLinks<HueService> {
  public hueServiceLinks: JSX.Element[] =
    this.service &&
    this.service.load_balancer.map(location => (
      <Menu.Item key={location.host}>
        <a target="_blank" rel="noreferrer noopener" href={`https://${location.host}:${location.port}`}>
          Hue UI
        </a>
      </Menu.Item>
    ));

  public links: JSX.Element[] = [
    // tslint:disable-next-line: jsx-key
    <Dropdown overlay={<Menu>{this.hueServiceLinks}</Menu>}>
      <a href="#" className="ant-dropdown-link">
        {/* eslint-disable-line */}
        Quick Links <Icon type="down" />
      </a>
    </Dropdown>,
  ];
}

export class HiveServiceLinks extends ServiceLinks<HiveService> {
  public links: JSX.Element[] = [0].map(index => (
    <h4 key={index} style={{ cursor: 'not-allowed', color: 'rgb(0,0,0,.45)', margin: 0 }}>
      (no links)
    </h4>
  ));
}

export class YarnServiceLinks extends ServiceLinks<YarnService> {
  public resourceManagerLinks: JSX.Element[] =
    this.service &&
    this.service.resource_manager.map(location => (
      <Menu.Item key={location.host}>
        <a target="_blank" rel="noreferrer noopener" href={`https://${location.host}:${location.port}`}>
          Resource Manager UI ({location.host})
        </a>
      </Menu.Item>
    ));

  public nodeManagerLinks: JSX.Element[] =
    this.service &&
    this.service.node_manager.map(location => (
      <Menu.Item key={location.host}>
        <a target="_blank" rel="noreferrer noopener" href={`https://${location.host}:${location.port}`}>
          Node Manager UI ({location.host})
        </a>
      </Menu.Item>
    ));

  public links: JSX.Element[] = [
    // tslint:disable-next-line: jsx-key
    <Dropdown
      overlay={
        <Menu>
          {this.resourceManagerLinks}
          {this.nodeManagerLinks}
        </Menu>
      }
    >
      <a href="#" className="ant-dropdown-link">
        {/* eslint-disable-line */}
        Quick Links <Icon type="down" />
      </a>
    </Dropdown>,
  ];
}

export class NavigatorServiceLinks extends ServiceLinks<NavigatorService> {
  public navigatorServiceLinks: JSX.Element[] =
    this.service &&
    this.service.navigator.map(location => (
      <Menu.Item key={location.host}>
        <a target="_blank" rel="noreferrer noopener" href={`https://${location.host}:${location.port}`}>
          Navigator UI
        </a>
      </Menu.Item>
    ));

  public links: JSX.Element[] = [
    // tslint:disable-next-line: jsx-key
    <Dropdown overlay={<Menu>{this.navigatorServiceLinks}</Menu>}>
      <a href="#" className="ant-dropdown-link">
        {/* eslint-disable-line */}
        Quick Links <Icon type="down" />
      </a>
    </Dropdown>,
  ];
}

export interface Cluster extends Statusable {
  name: string;
  cm_url: string;
  services: {
    hive: HiveService;
    hue: HueService;
    yarn: YarnService;
    mgmt: NavigatorService;
  };
}
