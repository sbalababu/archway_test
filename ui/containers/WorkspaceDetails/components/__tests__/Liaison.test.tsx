import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Liaison from '../Liaison';
import { Member } from '../../../../models/Workspace';

describe('Liaison', () => {
  it('renders correctly', () => {
    const data: Member = {
      name: 'hdfs',
      distinguished_name: 'rkaland',
      data: null,
      topics: null,
      applications: null,
    };
    const wrapper = shallow(<Liaison data={data} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
