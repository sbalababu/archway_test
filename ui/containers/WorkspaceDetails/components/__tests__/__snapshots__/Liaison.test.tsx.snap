// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Liaison renders correctly 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
    }
  }
>
  <div
    style={
      Object {
        "marginRight": 8,
        "textAlign": "right",
        "textTransform": "uppercase",
      }
    }
  >
    <div
      style={
        Object {
          "fontSize": 18,
          "fontWeight": 300,
        }
      }
    >
      OWNER
    </div>
    <a
      href="mailto:undefined"
      style={
        Object {
          "color": "rgb(67, 170, 139)",
          "textTransform": "uppercase",
        }
      }
    >
      hdfs
    </a>
  </div>
  <Icon
    style={
      Object {
        "fontSize": 40,
      }
    }
    theme="twoTone"
    twoToneColor="#D7C9AA"
    type="crown"
  />
</div>
`;
